### ISSUE TITLE
Visual Studio Code is no longer responding again

### ISSUE THREAD

**Issue by doronguttman** [Microsoft/vscode#6072]
Maybe reopen Visual Studio Code is no longer responding #160

[OK]
- VSCode Version: Version 1.0.0
  Commit fa6d0f03813dfb9df4589c30121e9fcffa8a8ec8
  Date 2016-04-13T14:08:36.599Z
  Shell 0.35.6
  Renderer 45.0.2454.85
  Node 4.1.1
- OS Version: Microsoft Windows [Version 6.1.7601]

Steps to Reproduce:
1. Open VSCode from a TypeScript app folder
2. Open a ts file
3. app freezes

**Comment by alexandrudima** [comment]
@doronguttman Can you please share the file that reproduces the problem? Is it freezing when opening a specific TypeScript file, or when opening a specific folder? Any other information that could help us reproduce the problem? [i.e. we're all editing TypeScript files all the time - VSCode is written in TypeScript]

**Comment by kmindi** [comment]
I have the same problem with files like [bootstrap.min.css](https://github.com/twbs/bootstrap/blob/06aa2a5bf24a31bc9a40cdf518eed5b15de9da30/dist/css/bootstrap.min.css)/bootstrap.min.js/font-awesome.min.css and similar files with few line breaks/long lines. 
(Don't know if those are the only files that cause it, but if someone can reproduce it with them that would be ok I think. Just copying the file in a new file freezes the editor already.)

(Version 1.1.1 Commit def9e32467 Shell 0.37.6, Renderer 49.0.2623.75, Node 5.10.0, OS Windows 7 64 bit, Extensions: https://marketplace.visualstudio.com/items?itemName=seanmcbreen.Spell
https://marketplace.visualstudio.com/items?itemName=TwentyChung.jsx
https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools
https://marketplace.visualstudio.com/items?itemName=austin.code-gnu-global

**Comment by alexandrudima** [comment]
@kmindi For 1.2.0 I've pushed two changes that target minified code files (very long lines). Would you mind [trying it out](https://code.visualstudio.com/insiders). For me, `bootstrap.min.css` opens fine -- interacting with it is not as smooth as with a non-minified file, but it causes no freezes for me anymore.

**Comment by kmindi** [comment]
@alexandrudima Seems to work now with the mentioned kind of files, thanks:) (referencing the pull request/commit to see what the cause/fix was would be interesting)

**Comment by alexandrudima** [comment]
Sure, it was ca2ca08c63c88f7c0bc2735a37a97ff3b8f600fc that targets all languages and 9326b1406b935d3188cb7afaec5ee78390b6d76a for JavaScript specific fix

**Comment by awaemmanuel** [comment]
I have this same issue. It freezes between work and there is nothing you can do unfreeze it. Only a restart works and then after a few lines of writing, it freezes again.

**Comment by vanloc0301** [comment]
I have same issues on Windows. It was horrid. I have much time to this issues.

**Comment by alexandrudima** [comment]
@awaemmanuel @vanloc0301 Please create new issues and describe in detail your project (it would be best if we could have access to it as well), otherwise please give us some hints / reproducible cases. A "me too" in this issue will not lead to a code change / improvement of VS Code.

**Comment by JosKrause** [comment]
**Just putting this here for information, it seems pointless to create a new issue where the only input would be: "Start Visual Studio Code => Lockup".**

I downloaded and installed the latest version, added a bunch of Extensions, then proceeded to try out my first TypeScript file, so I basically did CTRL+S to prompt a new file name, named it MyScript.ts.

Then proceeding to type or paste code, it completely locks up. In fact any file and extension will just lock up entirely. Now I cannot even uninstall packages, every startup simply locks up within a minute. I am not sure what I can do to "reproduce" the problem, as this simple "started to happen out of nowhere".

I am going to fully uninstall and reinstall Code to see if a clean state improves things, as the first days of using Code where without problems, and now without using for a week or so, it's completely unusable.

If there are any switches or parameters we can add to launching Code to help us debug this?

Just for argument sake, steps to reproduce the problem:

- Launch Visual Studio Code
- Try to type
- Try to open a file
- Try to save a blank file

All actions after opening the application result in a full lockup; now I can also postpone the uninstall and keep the current state if it helps debug the problem, otherwise going to reinstall on Monday.

**Comment by alexandrudima** [comment]
@JosKrause I'm sorry for this bad experience. Maybe you remember what extensions you have installed, there are some extensions that "patch" VSCode in "creative" ways. If you don't remember, then that's OK, check your `C:\Users\[...]\.vscode\extensions` folder. Perhaps you can rename it to `extensions2` and see if the problem still occurs. Also a bit sad, if VSCode is patched on your machine by an extension, then removing or renaming the extensions folder doesn't help, only a clean install would. You can also keep VSCode in this broken state and in parallel use a zip version from the downloads.

You can also check for errors in the console via `Help > Toggle Developer Tools`

@bpasero knows some good command line tricks to help troubleshoot.

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by eappell** [comment]
I had this issue, and the only way I could get VSCode to runwithout crashing was to rename the extensions folder to "extensions2", as @alexandrudima suggested (thanks!).  Then I dragged the extensions back into the "extensions" folder one at a time, restarting VSC each time until it crashed again.  The culprit, in my case, was Zignd.HTML-CSS-Class-Completion (https://github.com/Zignd/HTML-CSS-Class-Completion).  I had 24 other extensions, and none of them caused this issue.

**Comment by alexandrudima** [comment]
@eappell Please create an issue against the extension.

**Comment by rich25200** [comment]
@eappell same issue for me! Thanks for your tip.

**Comment by rafaelblink** [comment]
@alexandrudima removing C:\Users\[...]\.vscode\extensions worked perfectly, thanks bro.

**Comment by cino893** [comment]
Try this https://github.com/Microsoft/vscode/issues/12408#issuecomment-317745025
